### PR TITLE
Solve issue and fix bug

### PR DESCRIPTION
**Pull request by alexfilimon**
Issue #47 done - add ability to move generators. Bug #46 fixed - PaginableTableManager works improperly.

### CODE REVIEW

File: Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift [hunk 1]
```diff
@@ -323,6 +324,38 @@ extension BaseTableDataDisplayManager: UITableViewDelegate {
...
+
+        cellChangedPosition.invoke(with: (section: section, oldIndex: sourceIndexPath.row, newIndex: destinationIndexPath.row))
+    }
+
+    open func tableView(_ tableView: UITableView, shouldIndentWhileEditingRowAt indexPath: IndexPath) -> Bool {
+        return false
+    }
+
+    open func tableView(_ tableView: UITableView, editingStyleForRowAt indexPath: IndexPath) -> UITableViewCell.EditingStyle {
+        return .none
```
Comment (LastSprint) [inline Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift]: May we should configure style? 
If you want to move cell (with drag&drop) you should change editing style
May use `MovableGenerator` in this place?

Comment (alexfilimon) [reply Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift]: yes, that is good idea, but editing it is not about `MovableGenerator`. Maybe we should add `EditableGenerator` protocol to try to cast and get appropriate value?

Comment (LastSprint) [reply Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift]: Mmm, but in this case Movable require Editable :)
May be needs to add conformance between `MovableGenerator` and `EditableGenerator`?

Comment (alexfilimon) [reply Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift]: ok, I will add this

File: Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift [hunk 2]
```diff
@@ -323,6 +324,38 @@ extension BaseTableDataDisplayManager: UITableViewDelegate {
...
+        }
+        return false
+    }
+
+    open func tableView(_ tableView: UITableView, moveRowAt sourceIndexPath: IndexPath, to destinationIndexPath: IndexPath) {
+        guard sourceIndexPath.section == destinationIndexPath.section else {
+            return
+        }
+        let section = sourceIndexPath.section
+        var oldCellGenerators = cellGenerators[section]
```
Comment (LastSprint) [inline Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift]:
```suggestion
        var oldCellGenerators = self.cellGenerators[section]
```

File: Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift [hunk 3]
```diff
@@ -323,6 +324,38 @@ extension BaseTableDataDisplayManager: UITableViewDelegate {
...
+        let itemToMove = oldCellGenerators[sourceIndexPath.row]
+        oldCellGenerators.remove(at: sourceIndexPath.row)
+        oldCellGenerators.insert(itemToMove, at: destinationIndexPath.row)
+
+        let sectionGenerator = sectionHeaderGenerators[section]
+        removeAllGenerators(in: sectionGenerator)
```
Comment (LastSprint) [inline Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift]:
```suggestion
        self.removeAllGenerators(in: sectionGenerator)
```

File: Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift [hunk 4]
```diff
@@ -323,6 +324,38 @@ extension BaseTableDataDisplayManager: UITableViewDelegate {
...
+        addCellGenerators(oldCellGenerators, toHeader: sectionGenerator)
```
Comment (LastSprint) [inline Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift]:
```suggestion
        self.addCellGenerators(oldCellGenerators, toHeader: sectionGenerator)
```

File: Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift [hunk 5]
```diff
@@ -323,6 +324,38 @@ extension BaseTableDataDisplayManager: UITableViewDelegate {
...
+
+        cellChangedPosition.invoke(with: (section: section, oldIndex: sourceIndexPath.row, newIndex: destinationIndexPath.row))
```
Comment (LastSprint) [inline Source/BaseTableDataDisplayManager/BaseTableDataDisplayManager.swift]:
```suggestion
        self.cellChangedPosition.invoke(with: (section: section, 
                                                                         oldIndex: sourceIndexPath.row, 
                                                                         newIndex: destinationIndexPath.row))
```

**Pull request merged**